opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexCheck.java

```diff
@@ -87,19 +92,41 @@ public enum IndexCheckMode {
         DOCUMENTS
     }
 
+    /**
+     * Common exception for all check modes.
+     */
+    public static class IndexCheckException extends Exception {
```

Comment:
Moving these exception classes to its own file will help to reduce the complexity of this class

Reply:
fixed